### ISSUE TITLE
State/session 

### ISSUE THREAD

**Issue by CashCode** [php-telegram-bot/core#874]
Hello.
For chain chat with user we need sometimes to read session data, for example in test, when we have answer buttons (not inline, just buttons) and we need to go for second question, we have to get previous message from database, but it's not practical.
Should I write my own table to store user session or it has something like that? I see only commands. 
Regards.

**Comment by chuv1** [comment]
Check [SurveyCommand.php](https://github.com/php-telegram-bot/example-bot/blob/master/Commands/SurveyCommand.php) which utilizes `Conversation` class, that implements [Finite State Machine](https://en.wikipedia.org/wiki/Finite-state_machine), the thing you're talking about.

**Comment by noplanman** [comment]
@CashCode All clear then?

**Comment by CashCode** [comment]
Yes, but I had an issue when I tried to redefine route for command, I mean that there is default command to proceed for example query request (button click), I'd like to store state and proceed request with the same command.
